NorthstarDLL/squirrel.h

```diff
@@ -340,4 +221,203 @@ template <ScriptContext context> class SquirrelManager
 #pragma endregion
 };
 
+template <ScriptContext context> class SquirrelManager : public virtual SquirrelManagerBase
+{
+  public:
+#pragma region MessageBuffer
+	SquirrelMessageBuffer* messageBuffer;
+
+	template <typename... Args> SquirrelMessage schedule_call(std::string funcname, Args... args)
+	{
+		// This function schedules a call to be executed on the next frame
+		// This is useful for things like threads and plugins, which do not run on the main thread
+		FunctionVector function_vector;
+		SqRecurseArgs<context>(function_vector, args...);
+		SquirrelMessage message = {funcname, function_vector};
+		messageBuffer->push(message);
+		return message;
+	}
+
+	SquirrelMessage schedule_call(std::string funcname)
+	{
+		// This function schedules a call to be executed on the next frame
+		// This is useful for things like threads and plugins, which do not run on the main thread
+		FunctionVector function_vector = {};
+		SquirrelMessage message = {funcname, function_vector};
+		messageBuffer->push(message);
+		return message;
+	}
+
+	SQRESULT call(const char* funcname)
+	{
+		// Warning!
+		// This function assumes the squirrel VM is stopped/blocked at the moment of call
+		// Calling this function while the VM is running is likely to result in a crash due to stack destruction
+		// If you want to call into squirrel asynchronously, use `schedule_call` instead
+
+		if (!m_pSQVM || !m_pSQVM->sqvm)
+		{
+			spdlog::error(
+				"{} was called on context {} while VM was not initialized. This will crash", __FUNCTION__, GetContextName(context));
+		}
+
+		SQObject* functionobj = new SQObject();
```

Comment:
You allocate new memory, but should this be freed?
If it should, does the internal engine function 'pushSQObject' free this memory?

If it shouldn't, add a comment.